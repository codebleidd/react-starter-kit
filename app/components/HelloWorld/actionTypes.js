export const LOAD_HELLO = 'LOAD_HELLO'
