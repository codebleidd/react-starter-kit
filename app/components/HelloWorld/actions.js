import * as types from './actionTypes'

export function setHello (helloMessage) {
  return {
    type: types.LOAD_HELLO,
    helloMessage
  }
}
