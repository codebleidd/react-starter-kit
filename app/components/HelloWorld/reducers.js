import * as types from './actionTypes'

export default function helloMessage (state = '', action) {
  switch (action.type) {
    case types.LOAD_HELLO:
      return action.helloMessage
    default:
      return state
  }
}
