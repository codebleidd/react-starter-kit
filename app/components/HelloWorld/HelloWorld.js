import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { setHello } from './actions'

import style from './HelloWorld.scss'

export class HelloWorld extends React.Component {
  constructor (props, context) {
    super(props, context)

    this.state = { hello: 'Hello' }
  }

  componentDidMount () {
    const { dispatch } = this.props
    dispatch(setHello('HELLO'))
  }

  render () {
    const { hello } = this.state
    const { helloMessage } = this.props
    return (
      <div className={style.container}>
        { helloMessage || hello } World
      </div>
    )
  }
}

HelloWorld.propTypes = {
  helloMessage: PropTypes.string
}

function mapStateToProps (state) {
  return {
    helloMessage: state.helloMessage
  }
}

export default connect(mapStateToProps)(HelloWorld)
