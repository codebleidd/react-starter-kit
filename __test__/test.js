import React from 'react'
import { HelloWorld } from '../app/components/HelloWorld/HelloWorld'

describe('Hello world test', () => {
  it('render HelloWorld component', () => {
    const wrapper = shallow(<HelloWorld helloMessage='Hello' dispatch={jest.fn()} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should should be false', () => {
    expect(false).toBe(false)
  })
})
