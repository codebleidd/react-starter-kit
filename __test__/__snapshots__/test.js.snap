// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello world test render HelloWorld component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HelloWorld
    dispatch={[Function]}
    helloMessage="Hello"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Hello",
        " World",
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      "Hello",
      " World",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Hello",
          " World",
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        "Hello",
        " World",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
